platform_versioned_docs/version-24.1/data_studios/custom-envs.mdx

```diff
@@ -0,0 +1,129 @@
+
+Data Studios uses the [Wave][wave-home] service to build custom container template images.
+
+## Conda packages
+
+### Prerequisites
+
+- Wave must be configured. For more information, see [Wave containers][wave].
+ 
+ :::note
+ Enterprise deployments must either allow access to the public Wave server, or self-host their own Wave server.
+ :::
+
+### Conda package syntax {#conda-package-syntax}
+
+When adding a new data studio session, you can customize its configuration to install a list of Conda packages to the template image. The supported schema is identical to that used by the Conda `environment.yml` environment file. For more information on the Conda environment file, see [Creating an environment file manually][env-manually].
+
+```yaml title="Example environment.yml file"
+name: example-env
+channels:
+  - defaults
+  - conda-forge
+dependencies:
+  - python=3.7
+  - numpy
+  - pandas
+  - scikit-learn
+  - pip:
+    - matplotlib
+    - seaborn
+    - jupyter
+```
+
+To create a data studio with custom Conda packages, see [Add a data studio][add-ds].
+
+## Custom container template image {#custom-containers}
+
+For advanced use cases, you can build your own container template image.
+
+:::note
+Amazon Elastic Container Registry (ECR) is the only supported private container registry.
+:::
+
+### Prerequisites
+
+- Access to a container image repository, either a public container registry or a private Amazon ECR repository
+- A container template image
+
+### Dockerfile configuration {#dockerfile}
+
+For your custom template container image, you must use a Seqera-provided base image and include several additional build steps for compatibility with Data Studios.
+
+To create a data studio with a custom template image, see [Add a data studio][add-ds].
+
+#### Ports
+
+The container must use the value of the `CONNECT_TOOL_PORT` environment variable as the listening port for whatever interactive software that you include in your custom container.
+
+#### Signals
+
+Upon termination, the container's main process must handle the `SIGTERM` signal and perform any necessarily cleanup. After a 30 second grace period, the container receives the `SIGKILL` signal.
+
+#### Minimal Dockerfile
+
+The minimal Dockerfile includes directives to accomplish the following:
+
+- Pulls a Seqera-provided base image with prerequisite binaries
+- Installs and configures Seqera Fusion
+- Sets the container entry point
+
+Customize the following Dockerfile to include any additional software that you require:
+
+```docker title="Minimal Dockerfile"
+# Seqera base image
+FROM public.cr.seqera.io/platform/connect-client:0.7 as connect
+
+# 1. Add connect binary
+COPY --from=connect /connect-client /usr/bin/connect-client
+
+# 2. Install connect dependencies
+RUN /usr/bin/connect-client --install
+
+# 3. Configure connect as the entrypoint
+ENTRYPOINT ["/usr/bin/connect-client"]
```

Comment:
with the latest changes this line needs to be `ENTRYPOINT ["/usr/bin/connect-client", "--entrypoint"]`